packages/fx-core/src/plugins/resource/appstudio/utils/utils.ts

```diff
@@ -268,3 +268,24 @@ export function renderTemplate(manifestString: string, view: any): string {
   manifestString = Mustache.render(manifestString, view);
   return manifestString;
 }
+
+export class RetryHandler {
+  public static async Retry<T>(fn: () => Promise<T>): Promise<T | undefined> {
+    let retries = 10;
+    let response;
+    while (retries > 0) {
+      retries = retries - 1;
+      try {
+        response = await fn();
+        return response;
+      } catch (e: any) {
+        // Directly throw 404 error, keep trying for other status code e.g. 503 400
+        if (retries <= 0 || e.response?.status == 404) {
+          throw e;
+        } else {
+          await new Promise((resolve) => setTimeout(resolve, 5000));
```

Comment:
Will 5s be too long? If all retries failed, use will have to wait for 50s before getting the error.